### ISSUE TITLE
Alternate binding specification

### ISSUE THREAD

**Issue by finger563** [pthom/litgen#13]
I'm having difficulty adapting my project to use litgen, as the way it generates the python bindings leads to errors of the form:

```console
chosen constructor is explicit in copy-initialization
```

Here is the generated binding on the right (generated by litgen), and on the left is the original binding I hand-wrote:
![CleanShot 2024-07-27 at 22 30 56@2x](https://github.com/user-attachments/assets/a5be6b58-1621-4788-ab0b-e568ab26b82f)

For reference, here is the associated c++ code: 
![CleanShot 2024-07-27 at 22 37 57@2x](https://github.com/user-attachments/assets/9ec18742-f45f-45ea-ab57-c4c0617432db)

I've looked through the docs and the examples but I've yet to find any controls for how this part of the generated code can be configured - are there any controls that might help?

**Comment by pthom** [comment]
Hello,

Please try to use code blocks instead of screenshots, since it is easier to copy-paste. Also, please try to provide a minimal example that reproduces the error.

In your case, I suspect that `on_receive_callback` *might* be the member that creates the error. I cannot be certain, since I do not have access to the code, nor to the definition of receive_callback.

To write this class binding by hand use options.class_exclude_by_name__regex
To exclude the creation of a named ctor, use options.struct_create_default_named_ctor__regex

**Comment by finger563** [comment]
Thanks @pthom for the reply - I'm trying to not have to manually write bindings for these things, so I'm modifying my code so that litgen can handle it as best as possible.

Thanks for the tip on how to remove the creation of the named ctor, i'll try that!

I don't have a simply reproducible example yet, but i'll try to whip one up. In the meantime, this is the PR for the litgen work I'm doing at espp:

https://github.com/esp-cpp/espp/pull/291

**Comment by pthom** [comment]
> In the meantime, this is the PR for the litgen work I'm doing at espp: https://github.com/esp-cpp/espp/pull/291

I had a look at the [options](https://github.com/esp-cpp/espp/blob/968bbeb9ddf63739d585c5a9f27bb5bd8b929e12/lib/autogenerate_bindings.py). Nice!

**Comment by finger563** [comment]
@pthom thanks! yea i've been really liking your project and I'm hopeful that I can get it configured to automatically keep the python bindings complete and up to date, though right now I'm struggling because the way my code is written litgen doesn't seem to like parsing it :/

Related to the options, is there a different way I should be specifying template class structs, such as `template <typename T> RangeMapper { public: struct Config {...} }` so that it gets generated properly when i specialize the template container class? Right now my options which specify the same specializations for `RangeMapper::Config` as for `RangeMapper` do not seem to work.

Related to the original issue I posted - I found that it was specifically for the `std::chrono::duration<float>` default initialization I was specifying (which I was specifying as `0.5f`). If I change my default initialization value to be an explicit `std::chrono::duration<float>(0.5f)` then the error I was getting about copy-initialization went away.
